### ISSUE TITLE
flag warning if any outer has same name as feature

### ISSUE THREAD

**Issue by michaellilltokiwa** [tokiwa-software/fuzion#2985]
I think it is almost always a bad idea to name a feature same as one of its outers. We might want to show a warning if this is the case.

**Comment by fridis** [comment]
Right, using an outer feature's name for an inner feature is probably not a good idea. 
However, we get similar problems when inheriting features that happen to have the same name as an outer feature. 
What is important to me is to cause an error whenever there is an ambiguity, so any unqualified use of a feature `f` that exists at different outer levels should cause an error.
As for warnings: we do have some warnings, but generally, something should either be allowed or forbidden, there should be little in-between.

**Comment by simonvonhackewitz** [comment]
Can this be closed? I think ambiguity always causes an error.

**Comment by michaellilltokiwa** [comment]
@simonvonhackewitz When calling a feature, yes. But what about this:
```
ex is
  ex is
    ...
```
Problem is that 
```
ex is
  ex is
    say ex.this
```
```
-:3:9: error 1: Ambiguous outer feature reference in '.this'-expression
    say ex.this

Within feature: 'ex.ex'
Outer feature that was ambiguous: 'ex'
Outer features available: -- none! --

one error.
```

**Comment by michaellilltokiwa** [comment]
also this does not work:
```
ex is
  ex is
    say universe.ex.this
```
as well as this:
```
ex is
  ex is
    say universe.ex.ex.this
```
```
-:3:21: error 1: Qualifier expected for '.this' expression.
    say universe.ex.ex.this

Found expression 'ex' where a simple qualifier 'a.b.c' was expected

one error.
```

**Comment by fridis** [comment]
I agree that @michaellilltokiwa examples should work, in particular

- the negative `ex.this` should produce an error —as it does—, but it could suggest a solution `universe.ex.this` or `universe.ex.ex.this`
- the examples using `universe.ex...` should work.

So let's refocus this issue on that.
